src/spu/src/smart_stream/mod.rs

```diff
@@ -95,22 +14,10 @@ pub struct RecordsMemory {
 }
 
 impl RecordsMemory {
-    fn copy_memory_from(&self) -> Vec<u8> {
-        unsafe {
-            if let Some(data) = self
-                .memory
-                .data_unchecked()
-                .get(self.ptr as u32 as usize..)
-                .and_then(|arr| arr.get(..self.len as u32 as usize))
-            {
-                debug!(wasm_mem_len = self.len, "copying from wasm");
-                let mut bytes = vec![0u8; self.len as u32 as usize];
-                bytes.copy_from_slice(data);
-                bytes
-            } else {
-                vec![]
-            }
-        }
+    fn copy_memory_from(&self, store: &mut Store<()>) -> Result<Vec<u8>> {
```

Comment:
why is store needed?

Reply:
In the new API, a bunch of the wasmtime functions now take a `&mut Store<T>` where they didn't need to before. In this case, it is used in `self.memory.read(store, ...)`